indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,359 @@
+            var venues = {{ venues | tojson }};
+            var venueNames = {{ venue_names | tojson }};
+            var rooms = {{ rooms | tojson }};
+            var hiddenData = field.val() ? JSON.parse(field.val()) : {};
+            var latestUsedField = '';
+            var latestSelectedOption = {field: '', value: ''};
+
+            if (hiddenData.inheriting) {
+                usingDefault.prop('checked', hiddenData.inheriting);
+            }
+            if (hiddenData.address) {
+                address.val(hiddenData.address);
+            }
+            if (hiddenData.room_name) {
+                roomInput.val(hiddenData.room_name);
+            }
+            if (hiddenData.venue_name) {
+                venueInput.val(hiddenData.venue_name);
+            }
+
+            function toggleDefaultLocation(isChecked) {
+                var message = ('Using default location from: ' + {{ parent[1] | tojson }} +
+                               ' (' + {{ parent[0] | tojson }} + ')');
+                var text = isChecked ? message : 'Using custom location';
+                description.html(text);
+                venueInput.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+                roomInput.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+                address.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+
+                hiddenData.inheriting = isChecked;
+                field.val(JSON.stringify(hiddenData));
+            };
+
+            function reorderCategories(item) {
+                var groupOrder = [item.name];
+                for (var i = venueNames.length - 1; i >= 0; i--) {
+                    if (-groupOrder.indexOf(venueNames[i])) {
+                        groupOrder.push(venueNames[i]);
+                    }
+                }
+                Typeahead[roomInput.selector].options.groupOrder = groupOrder;
+            }
+
+            function updateVenueOnRoomChange(item) {
+                if (item.venue_id != hiddenData.venue_id) {
+                    var data = {item_name: item.group, venue_id: item.venue_id, room_id: item.id};
+                    venueInput.trigger('indico:click-on-option', data);
+                }
+            }
+
+            function updateResultsList(node, resultHtmlList) {
+                var resultsWrapper = node.siblings('.typeahead-results');
+                resultsWrapper.css('top', node.outerHeight() + 'px');
+                var justUse = node.parent().find('.just-use-option');
+                if (node.val().length > 0) {
+                    if (justUse.length === 0) {
+                        justUse = $('<div>').attr({
+                                'class': 'just-use-option'
+                            })
+                            .on('click', function() {
+                                node.trigger('indico:click-on-custom-option', {item_name: node.val()});
```

Comment:
Just checked the docs, didn't know about the object syntax. Looks actually cleaner than the list of unnamed arguments.